runhouse/utils.py

```diff
@@ -544,3 +547,53 @@ def format(self, output_type):
             self._display_title = True
 
         return system_color, reset_color
+
+
+####################################################################################################
+# Status collection utils
+####################################################################################################
+class ServletType(str, Enum):
+    env = "env"
+    cluster = "cluster"
+
+
+def get_gpu_usage(collected_gpus_info: dict, servlet_type: ServletType):
+
+    gpus = list(collected_gpus_info.keys())
+
+    # how we retrieve total_gpu_memory:
+    # 1. getting the first gpu usage of the first gpu un the gpus list
+    # 2. getting the first gpu_info dictionary of the specific gpu (we collected the gpu info over time)
+    # 3. get total_memory value (it is the same across all envs)
+    total_gpu_memory = collected_gpus_info[gpus[0]][0].get("total_memory")
```

Comment:
a little confused by the data structure here - why dod we need to parse gpus like this?

Reply:
my main point is why do we need to first collect this gpu info, and then iterate over again? is there a way to do this inline initially when constructing that data while still taking into account the servlet type?